pkg/cluster/generateconfig.go

```diff
@@ -100,6 +100,11 @@ func (m *manager) generateInstallConfig(ctx context.Context) (*installconfig.Ins
 		workerZones = []string{""}
 	}
 
+	SDNProvider := "OVNKubernetes"
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.SDNProvider != "" {
+		SDNProvider = string(m.doc.OpenShiftCluster.Properties.NetworkProfile.SDNProvider)
+	}
```

Comment:
I think in `pkg/api/openshiftcluster.go` we need to define constants for each SDN provider, simialr to what we do in the external representation:

```go
// SDNProvider constants.
type SDNProvider string

const (
	SDNProviderOVNKubernetes SDNProvider = "OVNKubernetes"
	SDNProviderOpenShiftSDN  SDNProvider = "OpenShiftSDN"
)
```

And we should replace hardcoded `"OVNKubernetes"` here with `api.SDNProviderOVNKubernetes` here.

Reply:
https://github.com/Azure/ARO-RP/pull/1571/commits/7cf26fefa84635caced03ddb15651f0635027f4f#diff-f9a5398ee035bbae6c18fc8468081bbbd9bab4d81504809a5589541ca2171aa7R198-R202

Constants

Reply:
https://github.com/Azure/ARO-RP/pull/1571/commits/7cf26fefa84635caced03ddb15651f0635027f4f#diff-2e16e231ac7cc84006ec5318e10e543d7d827adc2f36f3b7483d90665d6274abR103 

Used constant in default.